### PR TITLE
Removing nette/reflection [WIP]

### PR DESCRIPTION
**Pull request by dg**

### CODE REVIEW

File: src/Application/UI/Presenter.php [hunk 1]
```diff
@@ -504,7 +504,7 @@ public function formatLayoutTemplateFiles()
 		$name = $this->getName();
 		$presenter = substr($name, strrpos(':' . $name, ':'));
 		$layout = $this->layout ? $this->layout : 'layout';
-		$dir = dirname($this->getReflection()->getFileName());
+		$dir = dirname((new \ReflectionClass($this))->getFileName());
```
Comment (Majkl578) [inline src/Application/UI/Presenter.php]: Why not ReflectionObject?

Comment (dg) [inline src/Application/UI/Presenter.php]: Why ReflectionObject?

Comment (Majkl578) [inline src/Application/UI/Presenter.php]: Because `$this` is an instance, not a type name.

Comment (dg) [inline src/Application/UI/Presenter.php]: It works, ReflectionClass accepts objects.

Comment (Majkl578) [inline src/Application/UI/Presenter.php]: But ReflectionObject is semantically more correct, and also extends ReflectionClass.

Comment (enumag) [inline src/Application/UI/Presenter.php]: What's the difference between ReflectionClass and ReflectionObject anyway? I couldn't find anything worth mentioning in the docs.

Comment (dg) [inline src/Application/UI/Presenter.php]: I think that ReflectionObject::getProperties returns real object properties and ReflectionClass::getProperties returns default properties from a class.

**Comment by enumag** [comment]
@dg What about composer.json?

**Comment by dg** [comment]
In next version.

**Comment by Majkl578** [comment]
Why? It's not needed by this package so it should not be there, only in suggests section. If anyone was using it directly, they've been shooting themselves in the foot -- they should've had it in their composer.json.

**Comment by dg** [comment]
Because back compatibility

**Comment by Majkl578** [comment]
I don't get it. This package doesn't need that package anymore so it should just drop the dependency entirely. Noone should ever depend on a library indirectly through another dependency because it is just not what _dependency_ means.
Also it's ok to remove nette/security, but not nette/reflection? Doesn't make sense.

**Comment by dg** [comment]
It is directly used in code.

**Comment by enumag** [comment]
Even if this affects anyone its easily fixable by one line in composer.json. Its ok to put a minor BC break like that into 2.4 in my opinion.

**Pull request merged**